core/store/orm/orm.go

```diff
@@ -1488,6 +1488,47 @@ func toISO8601(t time.Time) string {
 		t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second(), t.Nanosecond(), tz)
 }
 
+const removeUnstartedTransactionsQuery = `
+WITH
+unstarted_job_runs AS (
+	DELETE FROM job_runs
+	WHERE status = 'unstarted'
+	RETURNING id, run_request_id
+),
+linked_run_requests AS (
+	DELETE FROM run_requests AS rr
+	WHERE rr.id IN (
+		SELECT run_request_id
+		FROM unstarted_job_runs
+	)
+),
+linked_task_runs AS (
+	DELETE FROM task_runs AS ts
+	WHERE ts.job_run_id IN (
+		SELECT id
+		FROM unstarted_job_runs
+	)
+	RETURNING id
+),
+linked_eth_txes AS (
+	DELETE FROM eth_txes AS et
+	USING eth_task_run_txes AS etr
+	WHERE etr.task_run_id IN (
+		SELECT id
+		FROM linked_task_runs
+	)
+	RETURNING id
+)
+DELETE FROM eth_tx_attempts AS eta
+USING linked_eth_txes AS let
+WHERE let.id = eta.eth_tx_id
```

Comment:
@samsondav, @se3000 I wonder whether for unstarted txes, there's any point to go through all these tables?

Reply:
Some of this can probably be removed since foreign keys are set with cascade, particular eth_tx_attempts.

I would advise checking all foreign keys.

Reply:
In a previous version of this PR, I did just that https://github.com/smartcontractkit/chainlink/pull/3362/files#diff-51b42c415a2523d9e9632bbde39ace6dR1441-R1444
I didn't find anything consistent so I decided to play it safe and write a query what makes it explicit what data is removed. I'm not sure how this interracts with cascade triggers: do triggers fire after the query?